fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -228,10 +228,14 @@ private void loadProtoExceptRecords(@Nonnull RecordMetaDataProto.MetaData metaDa
                 typeBuilder.setRecordTypeKey(LiteralKeyExpression.fromProtoValue(typeProto.getExplicitKey()));
             }
         }
+        PlanSerializationContext serializationContext = null;
         for (RecordMetaDataProto.PUserDefinedFunction function: metaDataProto.getUserDefinedFunctionsList()) {
+            if (serializationContext == null) {
```

Comment:
This is not needed in a refactored world. Strictly speaking, the plan serialization context is necessary state such that repeated sub plans/values/types do not get serialized repeatedly. It a little bit like dictionary encoding. I think this is not needed here.